### PR TITLE
fs/vfs/fs_ioctl.c: Add FIOCLEX/FIONCLEX support

### PR DESCRIPTION
**Pull request by sgysh**
## Summary

Add FIOCLEX/FIONCLEX support

## Impact

ioctl

## Testing

hifive1-revb:nsh
on QEMU

```
int main(int argc, FAR char *argv[])
{
  int fd = open("/dev/null", O_RDONLY);
  assert(fd != -1);

  int flags = fcntl(fd, F_GETFD);
  assert(flags != -1);
  assert(0 == flags & FD_CLOEXEC);

  int rc = ioctl(fd, FIOCLEX, NULL);
  assert(0 == rc);

  flags = fcntl(fd, F_GETFD);
  assert(flags != -1);
  assert(FD_CLOEXEC == flags & FD_CLOEXEC);

  rc = ioctl(fd, FIONCLEX, NULL);
  assert(0 == rc);

  flags = fcntl(fd, F_GETFD);
  assert(flags != -1);
  assert(0 == flags & FD_CLOEXEC);

  close(fd);

  printf("OK\n");
  return 0;
}
```

### CODE REVIEW

File: fs/vfs/fs_ioctl.c [hunk 1]
```diff
@@ -194,6 +194,12 @@ int nx_vioctl(int fd, int req, va_list ap)
                                nx_fcntl(fd, F_GETFL) & ~O_NONBLOCK);
               }
             break;
+          case FIOCLEX:
+            ret = nx_fcntl(fd, F_SETFD, FD_CLOEXEC);
```
Comment (xiaoxiang781216) [inline fs/vfs/fs_ioctl.c:198]: should we call F_GETFL and then or with FD_CLOEXEC? to avoid clear other flag we will add in the future.

Comment (sgysh) [reply fs/vfs/fs_ioctl.c:198]: I will change them as follows

```
nx_fcntl(fd, F_GETFD) | FD_CLOEXEC
nx_fcntl(fd, F_GETFD) & ~FD_CLOEXEC
```

Comment (sgysh) [reply fs/vfs/fs_ioctl.c]: I changed them in the commit https://github.com/apache/incubator-nuttx/pull/1800/commits/ac0c605c57e86445ac2aebcb6005b18bf268425a

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Pull request merged**